entc/gen/template/meta.tmpl

```diff
@@ -130,7 +130,7 @@ const (
 	{{ $name := $f.Validator }}
 	// {{ $name }} is a validator for the "{{ $f.Name }}" field enum values. It is called by the builders before save.
 	func {{ $name }}({{ $receiver }} {{ $enum }}) error {
-		switch {{ $receiver }} {
+		switch {{ $receiver }}{{ if $f.IsNonStringEnum }}.String(){{ end }} {
```

Comment:
```suggestion
		switch {{ $receiver }}{{ if $f.Type.Stringer }}.String(){{ end }} {
```

This is also OK because the validation already happened on the `schema/field` package.

Reply:
I think this is problematic in cases where you just provide string values (i.e. `Values("foo", "bar")`) due to:

https://github.com/adayNU/ent/blob/draft/non-string-enums/entc/gen/template/meta.tmpl#L118-L126

Because below typed constant values are used in the case statement: 

https://github.com/adayNU/ent/blob/draft/non-string-enums/entc/gen/template/meta.tmpl#L134

Reply:
> I think this is problematic in cases where you just provide string values (i.e. Values("foo", "bar")) due to:

Not sure why it's problematic. Can you provide a field example that will cause this template to produce invalid code?